### ISSUE TITLE
Chinese characters

### ISSUE THREAD

**Issue by lmm0591** [thlorenz/inline-source-map#6]
messy code with Chinese characters in source maps
like this:

        var generator = require('inline-source-map');
        var foo = '' + function foo () {
            var hello = '中文';
            var world = 'world';
            console.log('%s %s', hello, world);
        }

        var gen = generator()
            .addGeneratedMappings('foo.js', foo)
            .addSourceContent('foo.js', foo);

        console.log('inline mapping url:', gen.inlineMappingUrl());

        inline mapping url: //# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiIiwic291cmNlcyI6WyJmb28uanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUE7QUFDQTtBQUNBO0FBQ0E7QUFDQSIsInNvdXJjZXNDb250ZW50IjpbbnVsbF19

![ChineseCharacters](http://115.28.182.242/outher/1.png)

---
I modified the source code of inline-source-map in
/inline-source-map/index.js

![SourceCode](http://115.28.182.242/outher/2.jpg)

**Comment by thlorenz** [comment]
Should be fixed with 0.4 due to PR #8.
Please bug the browser-pack maintainer(s) to upgrade.

**Comment by terinjokes** [comment]
@thlorenz browser-pack uses `convert-source-map` which uses "~0.3.1". https://github.com/thlorenz/convert-source-map/blob/2ba9fa53177e87538acd35b6b2d2c929438a2ff5/package.json#L16

**Comment by thlorenz** [comment]
@terinjokes thanks, but that's just the dev dependency and is not used when the actual code runs.

**Comment by terinjokes** [comment]
Sorry I overlooked that. I'm not sure the connection between browser-pack
and inline-source-map then.

**Comment by thlorenz** [comment]
browser-pack [depends on combine-source-map](https://github.com/substack/browser-pack/blob/master/package.json#L11) which [ depends on inline-source-map](https://github.com/thlorenz/combine-source-map/blob/master/package.json#L15).

Should we update the version in there?

**Comment by terinjokes** [comment]
And now I think I see how I got to the wrong package while looking up
dependencies earlier. Very similar package names. ;)

**Comment by thlorenz** [comment]
Anything else we need to do here? Otherwise please close.

**Comment by csbun** [comment]
charset should use `=` instead of `:`, like this
```javascript
Generator.prototype.inlineMappingUrl = function () {
  var charset = this.opts.charset || 'utf-8';
  return '//# sourceMappingURL=data:application/json;charset=' + charset + ';base64,' + this.base64Encode();
};
```

see [Data URI scheme - Format](https://en.wikipedia.org/wiki/Data_URI_scheme#Format)

**Comment by thlorenz** [comment]
Is there an actual spec for this?
We shouldn't just follow a Wiki entry to change this.

Alternatively links to the code of different browsers that parses charset would be more useful than a wiki link.

**Comment by shootaroo** [comment]
It is [RFC 2397](https://tools.ietf.org/html/rfc2397), linked from the [wiki entry](https://en.wikipedia.org/wiki/Data_URI_scheme).

>        dataurl    := "data:" [ mediatype ] [ ";base64" ] "," data
>        mediatype  := [ type "/" subtype ] *( ";" parameter )
>        data       := *urlchar
>        parameter  := attribute "=" value

**Comment by thlorenz** [comment]
Fixed via #13
